File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/11.1 - For Loop More Exercises/01. Back To The Past .py
heritage = float(input())
goal_year = int(input())

age = 18

for year in range(1800, goal_year + 1):
    if year % 2 == 0:
        heritage -= 12000
        age += 1
    if year % 2 != 0:
        heritage -= 12000 + (50 * age)
        age += 1

if heritage >= 0:
    print(f"Yes! He will live a carefree life and will have {heritage:.2f} dollars left.")
else:
    print(f"He will need {abs(heritage):.2f} dollars to survive.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/12. УПРАЖНЕНИЕ ЦИКЛИ/For-Loop - More Exercises/01. Back To The Past.js
function backToThePast(input) {
    // Mask
    let money = Number(input.shift());
    let yearAim = input.shift();

    for (let year = 1800; year <= yearAim; year += 1) {
        if (year % 2 == 0) {
            money -= 12000;
        } else if (year % 2 != 0) {
            money -= (12000 + (50 * (year - 1800 + 18)));
        }
    }

    if (money >= 0) {
        console.log(`Yes! He will live a carefree life and will have ${money.toFixed(2)} dollars left.`);
    } else {
        console.log(`He will need ${Math.abs(money).toFixed(2)} dollars to survive.`);
    }
}

backToThePast([50000, 1802]);  // Should return:
// Yes! He will live a carefree life and will have 13050.00 dollars left.

backToThePast([100000.15, 1808]);  // Should return:
// He will need 12399.85 dollars to survive.